descartes_core/src/planning_graph.cpp

```diff
@@ -805,36 +878,32 @@ bool PlanningGraph::calculateJointSolutions()
     joint_solutions_map_.clear();
   }
 
-  // for each TrajectoryPt, get the available joint solutions
-  for (std::map<TrajectoryPt::ID, CartesianPointInformation>::iterator trajectory_iter = cartesian_point_link_->begin();
-      trajectory_iter != cartesian_point_link_->end(); trajectory_iter++)
-  {
-    // TODO: copy this block to a function that can be used by add and modify
-    /*************************/
-    std::list<TrajectoryPt::ID> *traj_solutions = new std::list<TrajectoryPt::ID>();
-    std::vector<std::vector<double> > joint_poses;
-    trajectory_iter->second.source_trajectory_.get()->getJointPoses(*robot_model_, joint_poses);
+  // Copy trajectory pt ptrs to a vector to be solved in parallel
+  std::vector<TrajectoryPtPtr> trajectory;
+  trajectory.reserve(cartesian_point_link_->size());
 
-    TrajectoryPt::ID tempID = trajectory_iter->first;
-    ROS_INFO("CartID: %s: JointPoses count:%i", boost::uuids::to_string(tempID).c_str(), (int)(joint_poses.size()));
+  for (const auto& kv : *cartesian_point_link_) // could also use lambda and std::for_each
+  {
+    trajectory.push_back(kv.second.source_trajectory_);
+  }
+  
+  // Compute joint solutions for trajectory pts
+  VecJointSolutions all_joint_solutions = parallelCalculateJointSolutions(trajectory, *robot_model_, max_threads_);
 
-    if (joint_poses.size() == 0)
```

Comment:
This error message should probable be kept since it is useful for troubleshooting.